iAM-legacy/Simulation/Investments.cs

```diff
@@ -234,15 +234,17 @@ public void LoadBudgets()
         /// <param name="priority"></param>
         /// <param name="limits">Limits where treatments are split over multiple years</param>
         /// <returns></returns>
-        public string IsBudgetAvailable(float fAmount, String strBudget, String strYear, Hashtable hashAttributeValue,Priorities priority, List<ISplitTreatmentLimit> limits, string budgetLimitType, out string budgetHash, out ISplitTreatmentLimit splitTreatmentLimit)
+        public string IsBudgetAvailable(float fAmount, string strBudget, String strYear, Hashtable hashAttributeValue,Priorities priority, List<ISplitTreatmentLimit> limits, string budgetLimitType, out string budgetHash, out ISplitTreatmentLimit splitTreatmentLimit, out string reasonNoBudget)
```

Comment:
A budget type class should be created to handle budget state and behaviors. Please create the appropriate object to handle this data and use it here.